tools/buildsys/src/args.rs

```diff
@@ -0,0 +1,143 @@
+/*!
+
+These structs provide the CLI interface for buildsys which is called from Cargo.toml and accepts all
+of its input arguments from environment variables.
+
+!*/
+
+use clap::{Parser, Subcommand};
+use std::path::PathBuf;
+
+/// A tool for building Bottlerocket images and artifacts.
+#[derive(Debug, Parser)]
+pub(crate) struct Buildsys {
+    #[command(subcommand)]
+    pub(crate) command: Command,
+}
+
+#[derive(Subcommand, Debug)]
+pub(crate) enum Command {
+    BuildPackage(BuildPackageArgs),
+    BuildVariant(BuildVariantArgs),
+}
+
+/// Arguments common to all subcommands.
+#[derive(Debug, Parser)]
+pub(crate) struct Common {
+    #[arg(long, env = "BUILDSYS_ARCH")]
+    pub(crate) arch: String,
+
+    #[arg(long, env = "BUILDSYS_OUTPUT_DIR")]
+    pub(crate) output_dir: PathBuf,
+
+    #[arg(long, env = "BUILDSYS_ROOT_DIR")]
+    pub(crate) root_dir: PathBuf,
+
+    #[arg(long, env = "BUILDSYS_STATE_DIR")]
+    pub(crate) state_dir: PathBuf,
+
+    #[arg(long, env = "BUILDSYS_TIMESTAMP")]
+    pub(crate) timestamp: String,
+
+    #[arg(long, env = "BUILDSYS_VERSION_FULL")]
+    pub(crate) version_full: String,
+
+    #[arg(long, env = "CARGO_MANIFEST_DIR")]
+    pub(crate) cargo_manifest_dir: PathBuf,
+
+    #[arg(long, env = "TLPRIVATE_SDK_IMAGE")]
+    pub(crate) sdk_image: String,
+
+    #[arg(long, env = "TLPRIVATE_TOOLCHAIN")]
+    pub(crate) toolchain: String,
+}
+
+impl Common {
+    pub(crate) fn rerun_for_envs() {
+        [
+            "BUILDSYS_ARCH",
+            "BUILDSYS_OUTPUT_DIR",
+            "BUILDSYS_ROOT_DIR",
+            "BUILDSYS_VERSION_FULL",
+            "CARGO_MANIFEST_DIR",
+            "TLPRIVATE_SDK_IMAGE",
+            "TLPRIVATE_TOOLCHAIN",
+        ]
+        .iter()
+        .for_each(|&var| rerun_for_env(var));
+    }
+}
+
+/// Build RPMs from a spec file and sources.
+#[derive(Debug, Parser)]
+pub(crate) struct BuildPackageArgs {
+    #[arg(long, env = "BUILDSYS_PACKAGES_DIR")]
+    pub(crate) packages_dir: PathBuf,
+
+    #[arg(long, env = "BUILDSYS_VARIANT")]
+    pub(crate) variant: String,
+
+    #[arg(long, env = "BUILDSYS_SOURCES_DIR")]
+    pub(crate) sources_dir: PathBuf,
+
+    #[arg(long, env = "CARGO_PKG_NAME")]
+    pub(crate) cargo_package_name: String,
+
+    #[command(flatten)]
+    pub(crate) common: Common,
+}
+
+impl BuildPackageArgs {
+    pub(crate) fn rerun_for_envs() {
+        Common::rerun_for_envs();
+        [
+            "BUILDSYS_PACKAGES_DIR",
+            "BUILDSYS_SOURCES_DIR",
+            "BUILDSYS_VARIANT",
```

Comment:
This will trigger a rebuild of all packages whenever the variant changes. We need to continue using the selective logic in main.rs instead.
```suggestion
```